File: DSA-DAILY/29-07-2025/2411. Smallest Subarrays With Maximum Bitwise OR@leetcode-med.py
class Solution:
    def smallestSubarrays(self, nums):
        n = len(nums)
        ans = [1] * n  # initialize answer array with 1s
        for i in range(n):
            x = nums[i]
            for j in range(i - 1, -1, -1):
                if (nums[j] | x) == nums[j]:
                    break
                nums[j] |= x
                ans[j] = i - j + 1
        return ans
if __name__ == "__main__":
    sol = Solution()
    nums = [1, 2, 3, 4]
    print(sol.smallestSubarrays(nums))

File: DSA-DAILY/29-07-2025/2411. Smallest Subarrays With Maximum Bitwise OR@leetcode-med.cpp
class Solution {
public:
    vector<int> smallestSubarrays(vector<int>& nums) {
        int n=nums.size();
        vector<int> ans(n); 
        for(int i=0;i<n;i++){
            int x=nums[i];
            ans[i]=1;
            for(int j=i-1;j>=0&&(nums[j]|x)!=nums[j];--j){
                nums[j]|=x;
                ans[j]=i-j+1;
            }
        }
        return ans;
    }
};
int main(){
  Solution sol;
  int n;
  cout<<"Enter size of array:";
  cin>>n;
  cout<<"Enter Array:"
  vector<int> n;
  for(int i=0;i<n;i++){
    cin>>a[i];
  }
  smallestSubarrays(nums);
  return 0;
}
